### PR TITLE
Py3 fixes4

### PR DESCRIPTION
**Pull request by ilvalle**
- Updated pysimplesoap although it is not passing py3 web2py tests, due to https://github.com/pysimplesoap/pysimplesoap/issues/107
- few py3 fixes

### CODE REVIEW

File: applications/welcome/controllers/default.py [hunk 1]
```diff
@@ -20,6 +20,15 @@ def index():
     response.flash = T("Hello World")
     return dict(message=T('Welcome to web2py!'))
 
+def grid():
```
Comment (niphlod) [inline applications/welcome/controllers/default.py]: ahem ?!

Comment (ilvalle) [inline applications/welcome/controllers/default.py]: I forgot to remove it

Comment (niphlod) [inline applications/welcome/controllers/default.py]: no prob... scouring through commits kinda became my second job ^_^'

**Comment by ilvalle** [comment]
test_web works in PY3 with pydal v16.03, however it raises an exception with the current pydal master. @gi0baro do you have an idea? see the the following https://travis-ci.org/web2py/web2py/jobs/139535525

**Comment by gi0baro** [comment]
@ilvalle as far as I can understand somewhere the code is putting a `str` value for the `pool_size` instead of an `int`:

```
Traceback (most recent call last):
  File "/home/travis/build/web2py/web2py/gluon/main.py", line 468, in wsgibase
    BaseAdapter.close_all_instances('commit')
  File "/home/travis/build/web2py/web2py/gluon/packages/dal/pydal/connection.py", line 117, in close_all_instances
    db._adapter.close(action)
  File "/home/travis/build/web2py/web2py/gluon/packages/dal/pydal/connection.py", line 94, in close
    if len(pool) < self.pool_size:
TypeError: unorderable types: int() < str()
```

**Comment by ilvalle** [comment]
Got it, it's here https://github.com/web2py/web2py/blob/master/applications/welcome/private/appconfig.ini#L17
I committed an updated version and now all tests passed.

**Pull request merged**